homework/zadanie4_five_philosophers.cpp

```diff
@@ -8,97 +8,102 @@
 
 using namespace std::chrono_literals;
 
-std::mutex mtx;
 
+class CFork{
+public:
+    mutable std::mutex mtx_fork_;
+};
+
+ 
 class CTable{
 
+    int numbers_of_forks_;
+
 public:
-    std::vector<bool> forks_pool;
+    std::vector<CFork*> forks_pool_;
 
-    CTable(int number_of_forks){
-        for (size_t i=0; i< number_of_forks; i++){
-            forks_pool.push_back(false);
+    CTable(int number_of_forks) : numbers_of_forks_(number_of_forks){
+        for (size_t i=0; i < numbers_of_forks_; i++){
+            forks_pool_.push_back(new CFork());
+        }
+    }
+
+    ~CTable(){
+        for (int i = 0; i < numbers_of_forks_; i++){
+            delete forks_pool_[i];
         }
     }
 };
 
 class CPhilosoph{
 
-public:
     std::string name_;
-    std::string output;
-    int number_;
-    CTable table_;
-    int left_fork_;
-    int right_fork_;
-
-    std::chrono::high_resolution_clock::time_point start_of_live_, end_of_live_;
-    size_t time=0;
+    std::string output_;
+    CTable& table_;
```

Comment:
A jakby to zamodelować w drugą stronę i to CTable trzyma jeszcze `std::vector<CPhilosoph>`?

Reply:
Przemodelowalem program, powinien bardziej sie spodobac niz wczesniejsza wersja.